llvm/lib/Transforms/Vectorize/VPlan.cpp

```diff
@@ -424,13 +450,19 @@ void VPBasicBlock::connectToPredecessors(VPTransformState &State) {
   if (ParentLoop && !State.LI->getLoopFor(NewBB))
     ParentLoop->addBasicBlockToLoop(NewBB, *State.LI);
 
+  auto Preds = to_vector(getHierarchicalPredecessors());
+  if (VPBlockUtils::isHeader(this, State.VPDT)) {
+    // There's no block for the latch yet, connect to the preheader only.
+    Preds = {Preds[0]};
+  }
```

Comment:
Can alternatively do:
```suggestion
    Preds = getPredecessors()[0];
  } else {
    Preds = to_vector(getPredecessors());
  }
```